fedimint-api/src/db/mod.rs

```diff
@@ -13,17 +13,22 @@ pub mod mem_impl;
 
 pub use tests::test_db_impl;
 
-pub trait DatabaseKeyPrefixConst {
-    const DB_PREFIX: u8;
+pub trait DbKeyValue: DbKeyPrefix {
     type Key: DatabaseKey;
     type Value: DatabaseValue;
 }
 
-pub trait DatabaseKeyPrefix: Debug {
+// This trait is implemented for database key prefix enums.
+pub trait DbKeyPrefix {
+    type KeyPrefixType: std::convert::Into<u8>;
+    const KEY_PREFIX: Self::KeyPrefixType;
+}
+
+pub trait DbKeyToBytes: Debug {
```

Comment:
I would name it `DBKey`. The that it converts "to bytes" is kind of less important. The main thing is that "it functions as a database key".

Reply:
There is a trait called `DatabaseKey` defined below. I think that sorting out and simplifying traits should be a separate PR.

I renamed to `DbKeyToBytes` from `DatabaseKeyPrefix` in this instance because I needed to commandeer the "prefix" name/concept.